### DISCUSSION TITLE
Latest paste version?

### DISCUSSION THREAD

**Discussion by robyn-bautista** [twilio-labs/paste#3177] [General] [answered]
Hey team, a few questions regarding latest paste updates:

• What version is the newly released paste? (aka what is the sunseted older version)
• When was the newest release?
• Is the current messaging that designers should still use the older [SUNSET] components in figma?

**Comment by TheSisb** [comment]
Hi @robyn-bautista,

> What version is the newly released paste? (aka what is the sunseted older version)

The latest version of the Paste packages can be found in a few different ways:
- In the [header of the Paste Core package page](https://paste.twilio.design/core)
- Listed on our [installation page](https://paste.twilio.design/introduction/for-engineers/manual-installation#paste-packages)
- By reviewing the [Paste changelog ](https://paste.twilio.design/core/changelog)
- Or [directly on NPM](https://www.npmjs.com/package/@twilio-paste/core)

> When was the newest release?

Good question! We are constantly releasing new versions of Paste as part of our [continuous delivery (CD) methodologies](https://continuousdelivery.com/). We recommend downstream teams to find a cadence that is comfortable for them to allocate towards updating Paste. Since we follow [semantic versioning (semver)](https://semver.org/) strictly, most updates are relatively painless. The only time consideration is required is when the "major" version changes (the left-most number before the `.`).

Specifically however, you can see [the dates of individual releases on NPM](https://www.npmjs.com/package/@twilio-paste/core?activeTab=versions).

**Reply by serifluous** [reply]
For more detail, 1 Twilio visual updates started being applied with [version 17.5.0](https://paste.twilio.design/core/changelog#17-5-0).

For when teams should switch, our blanket recommendation is that teams should swap to the newest versions and to the [NEW] Paste components library once the [Navigation UI Kit is released, planned for end of Q2](https://paste.twilio.design/roadmap) and your team has implemented it. Ultimately, though, it's a discussion that each product team should have about when it makes sense for y'all to update, and what the customer impact/risk is to swap before our recommended timeline.

You can keep using the [SUNSET] library in Figma. But if you're looking to future-proof some of your design work, we're also recommending designers use the [NEW] library for anything shipping in Q3+.

For the Flex team, you can read our team-specific guidance here: https://docs.google.com/document/d/1RACJwSG78YJtKMDgFY-VnbUzYyDqTae8nc3e-Y9a9Ho/edit#heading=h.wavpztde57db

**Comment by theryankennedy** [comment]
The latest versions of Aion UI components are using the latest Paste components in v19+, keeping more or less up to date with the latest releases.  The only Aion UI team that cares about Navigation is the one building the navigation, the majority are creating other components.  

Should Aion UI teams be downgrading components to specific versions?  Is it design tokens and themes that we need to be versioning in a specific way?   What is [SUNSET]?

**Comment by robyn-bautista** [comment]
it seems like some teams in Aion have been using the flex theme, but we'll plan to switch to "default" and wait to change to "twilio" if that is still the recommendation @serifluous @TheSisb

**Reply by SiTaggart** [reply]
There is technically no such thing as the Flex theme. Never has been, it was a relic of a past decision.

Read more here: https://github.com/twilio-labs/paste/discussions/3026#discussioncomment-4925084

**Comment by SiTaggart** [comment, marked as answer]
Hi hi

Let me try and clear up the confusion here. No downgrading of versions is necessary.

With updating the design language and rolling that, not every team is going to be able to do so at the same pace. In code, we have a couple of mechanisms to cope with that;

- We can version code so that people can upgrade as they wish. This is especially useful for Flex, as pushing these updates to customers might cause problems.
- We can create a new theme for those teams who want to be using the latest code, but aren't quite ready to switch to the updated design language. We've been coupling that option with the release of the navigation UI kit, so that the new design language has an appropriate frame within which to reside in.

We aren't so blessed in Figma, but we need to support the same use cases. So the [SUNSET] Figma Component Library is just an older version of the components before we started updating the design language. That way, product teams who are _unable_ to update their code version can continue to design features with a Figma library that matches the Code version.

For those who can update their code version, you can design using the [SUNSET] Figma library as you would normally, as those components will match the code. You will just be missing out on any new components we'll ship _after_ version 19, in code.
 
In the same way we support theming in code, we support theming in Figma. For those who aren't ready to transition to the new design language, you can use the default and dark theme in code and Figma. For those who are ready to try out the new look and feel, switch to using the Twilio and Twilio-dark themes, applied to the latest version of the component libraries.

That's a lot of words, and seamless, low impact migrations are hard, but we've given you a bunch of different tools to manage your products forward progress at your own pace, as you decide as a team.

I'll try and outline the above in a table:

| Design Language | Core Version Number | Code theme | Supported Figma Component Library | Supported Figma Theme Library |
| --------------- | ------------------- | ---------- | --------------------------------- | ------------------------------ |
| Current | <= v17.4 | default, <br />dark | [SUNSET] Paste Components | Paste Default Theme, <br />Paste Dark Theme |
| Current | > v17.4 | default, <br />dark | [SUNSET] Paste Components | Paste Default Theme, <br />Paste Dark Theme |
| One Twilio | >= v19.0.0 | twilio, <br />twilio-dark | [NEW] Paste Components | Paste Twilio Theme, <br />Paste Twilio Dark Theme |

**Reply by theryankennedy** [reply]
Thanks Simon, this is very helpful.

**Reply by theryankennedy** [reply]
If we are using the v19 paste core in code, and setting the theme to `default`, what theme are we seeing?  Is it `twilio`?  I bet I could look at the code to find this out.

**Reply by SiTaggart** [reply]
Nope, you're seeing the default theme.

**Reply by SiTaggart** [reply]
TL;DR The Code does exactly what it says on the tin, v18 and onwards. Figma doesn't do versioning well.

**Reply by robyn-bautista** [reply]
super helpful Simon, thank you!
